src/BinaryParsers/PEBinary/ProgramDatabase/LinkerCommandLine.cs

```diff
@@ -24,82 +24,160 @@ internal struct LinkerCommandLine
             bool order = false;
-            bool explicitlyEnabled = false;
-            bool explicitlyDisabled = false;
+            bool profile = false;
+
             foreach (string argument in ArgumentSplitter.CommandLineToArgvW(commandLine))
             {
                 if (!CommandLineHelper.IsCommandLineOption(argument))
                 {
                     continue;
                 }
 
-                // There are multiple /debug options so use StartsWith
-                if (ArgumentStartsWith(argument, "debug"))
-                {
-                    debugSet = true;
-                }
-                else if (ArgumentEquals(argument, "opt:ref"))
+                // There are multiple /debug options so use StartsWith.  There is also /debugtype: so we must be careful not
+                // to over-match against that.  /debug:none must go first.
+                if (ArgumentEquals(argument, "debug:none"))
                 {
-                    optRef = true;
+                    debugExplicit = false;
                 }
-                else if (ArgumentEquals(argument, "opt:icf"))
+                else if (ArgumentStartsWith(argument, "debug:") || ArgumentEquals(argument, "debug"))
                 {
-                    optIcf = true;
+                    debugExplicit = true;
                 }
-                else if (ArgumentEquals(argument, "opt:lbr"))
+                else if (ArgumentStartsWith(argument, "opt"))
                 {
-                    optLbr = true;
+                    // "The /OPT arguments may be specified together, separated by commas. For example, instead of
+                    //  /OPT:REF /OPT:NOICF, you can specify /OPT:REF,NOICF."
+                    // https://docs.microsoft.com/cpp/build/reference/opt-optimizations?view=msvc-170#remarks
+                    //
+                    // Make sure to match the more-specific noX before X because we are using contains and both would
+                    // match the enabled version.
+                    if (argument.Contains("noref", System.StringComparison.OrdinalIgnoreCase))
+                    {
+                        optimizeReferencesExplicit = false;
+                    }
+                    else if (argument.Contains("ref", System.StringComparison.OrdinalIgnoreCase))
+                    {
+                        optimizeReferencesExplicit = true;
+                    }
+
+                    if (argument.Contains("noicf", System.StringComparison.OrdinalIgnoreCase))
+                    {
+                        COMDATFoldingExplicit = false;
+                    }
+                    else if (argument.Contains("icf", System.StringComparison.OrdinalIgnoreCase))
+                    {
+                        COMDATFoldingExplicit = true;
+                    }
+
+                    if (argument.Contains("nolbr", System.StringComparison.OrdinalIgnoreCase))
+                    {
+                        optLbr = false;
```

Comment:
>optLbr [](http://example.com/codeflow?start=24&length=6)

do we sufficiently explain the long branch reference options? I'm not sure. i.e., it'd be nice if optLbr, order, profile, etc., had more descriptive names and/or comment at variable declaration that provide a bit more info.

Reply:
Added a comment to where the variables are defined at the top as a group.